### PR TITLE
feat: update language around commitments

### PR DESCRIPTION
**Pull request by tplooker**
This PR attempts to clarify the usage of the term commitment within the spec in general. I have added in general definitions for this and associated concepts and then also provide definition around things like recovery commitment and update commitment.

### CODE REVIEW

File: docs/spec/operations.md [hunk 1]
```diff
@@ -23,28 +23,28 @@ Sidetree Methods MAY rely on the DID Document data model when verifying protocol
...
+2. Generate a [recovery commitment](#recovery-commitment) using the defined [recovery commitment scheme](#recovery-commitment-scheme) and public key of the generated [recovery key pair](#recovery-key-pair).
+3. Generate an [operation commit value](#operation-commit-value), let the result be known as the _Update Operation Commit Value_.
+4. Generate an [operation commitment](#operation-commitment) using the _Update Operation Commit Value_ as the [operation commit value](#operation-commit-value), let the result be known as the _Update Operation Commitment_.
 5. Generate an encoded representation of the following object using the implementation's [`DATA_ENCODING_SCHEME`](#data-encoding-scheme), herein referred to as the [_Create Operation Delta Object_](#create-delta-object){ id="create-delta-object" }:
     ```json
     {
       "patches": [ PATCH_1, PATCH_2, ... ],
-      "update_commitment": HASH_OF_UPDATE_COMMITMENT_VALUE
+      "update_commitment": Update Operation Commitment
```
Comment (OR13) [inline docs/spec/operations.md:34]: I'd prefer to see a real world example here,

Comment (tplooker) [reply docs/spec/operations.md:34]: Yeah I agree in general I think real world examples here is a good idea

**Comment by OR13** [comment]
Make sure to address https://github.com/decentralized-identity/sidetree/issues/737 as well.

File: docs/spec/operations.md (hunk 1 above)
Comment (OR13) [reply docs/spec/operations.md:34]: @csuwildcat @thehenrytsai can you please pull real world values for this and share them on https://github.com/decentralized-identity/sidetree/issues/737

**Comment by tplooker** [comment]
Have updated to reflect the changes, have not done update or recover refactor yet, still looking for general feedback around the usage of the terms

**Comment by OR13** [comment]
@tplooker looks good, lets get this out of draft and ready to merge asap.

**Comment by tplooker** [comment]
@OR13 if you are happy with this direction then I will proceed with refactoring the remaining operations to this language, namely update recover and deactivate

**Comment by OR13** [comment]
@tplooker I am lets gooooo!

**Comment by OR13** [comment]
@tplooker 19 day old PR, can we get this out of draft mode, and merged BEFORE we cut a version ... @csuwildcat ...

**Comment by tplooker** [comment]
@OR13 perhaps to prevent this PR from getting to large we should review this PR as is?

**Comment by tplooker** [comment]
Has been updated out of draft status accordingly

File: docs/spec/operations.md (hunk 1 above)
Comment (OR13) [reply docs/spec/operations.md:32]: don't bother, we can fix this stuff later.

**Comment by OR13** [comment]
23 days open 1 approval. merging... but with a deep sense of shame.

**Pull request merged**